m4/pbs_patch_libtool.m4

```diff
@@ -0,0 +1,55 @@
+
+#
+# Copyright (C) 1994-2020 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of both the OpenPBS software ("OpenPBS")
+# and the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# OpenPBS is free software. You can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version.
+#
+# OpenPBS is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public
+# License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# PBS Pro is commercially licensed software that shares a common core with
+# the OpenPBS software.  For a copy of the commercial license terms and
+# conditions, go to: (http://www.pbspro.com/agreement.html) or contact the
+# Altair Legal Department.
+#
+# Altair's dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of OpenPBS and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair's trademarks, including but not limited to "PBS™",
+# "OpenPBS®", "PBS Professional®", and "PBS Pro™" and Altair's logos is
+# subject to Altair's trademark licensing policies.
+
+#
+
+AC_DEFUN([PBS_AC_PATCH_LIBTOOL], [
+	AC_CONFIG_COMMANDS([patch-libtool], [
+		AS_IF([! grep '[[-]]fsanitize=\*' libtool 2>&1 >/dev/null], [
+			AC_MSG_NOTICE([patching libtool to support -fsanitize])
+			mv libtool libtool.orig
+			sed -e 's/\(-fuse-linker-plugin\)\([[|)]]\)/\1|-fsanitize=\*\2/' \
```

Comment:
Hi @toonen  this sed line does not seem to be adding -fsanitize on my copy of libtool. Can you check further?

Reply:
I'm hoping the changes in fa6afda76d973f41547eaa8ee6cef7e163173ea0 will resolve this; however, if it is still failing, please send me the contents of `libtool.patched.err` and the specific OS and libtool versions you are using.

Reply:
Does not seem to work still, here is my container info:
[root@pbs pbspro]# libtool --version
libtool (GNU libtool) 2.4.2
Written by Gordon Matzigkeit <gord@gnu.ai.mit.edu>, 1996

Copyright (C) 2011 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
[root@pbs pbspro]# uname -a
Linux pbs 4.19.76-linuxkit #1 SMP Tue May 26 11:42:35 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
[root@pbs pbspro]# cat /etc/os-release 
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

I ran this command from your last commit:
sed 's/\(-pg\)\([[|)]]\)/\1|-fsanitize=\*\2/' libtool >libtool.patched 2>libtool.patched.err

and found a empty (0 bytes) libtool.patched.err file. The libtool.patched file is exactly same as libtool. This means sed passed through this file without making any changes.